### ISSUE TITLE
[enhancement] Latex export

### ISSUE THREAD

**Issue by scopeofaperture** [Proektsoftbg/Calcpad#73]
It would be brilliant to have latex export options. As Word is compatible with latex notation and I assume you already use it, the base structure to achieve this might already be there. I thought of two useful "modes" for this.

- just export the equations as plain latex.
- export each line in a \newcommand{\name}[count]{definition} so the file can just be included, and every line just added by executing it

The problem with the second approach would be to keep track of line naming/comandnname if a new line is inserted before or after a line. But if this is solved, you could write a Latex document and change values all over the finished document just by exporting again.
That way, calcpad doesn't have to be a text editor and calculator at once. Not a programmer but just wanted to share this Idea. I enjoy using Calcpad without also.

**Comment by Proektsoftbg** [comment]
@scopeofaperture Thank you for your proposition! I agree that LaTeX export (and why not import) would be an useful feature. I have thought about this before, but could not find time to implemented. Also, was now quite sure if this would be a wanted feature. I will add this to the wishlist for the near future.

**Comment by sononicola** [comment]
Definitely a feature I like to see!
If you plan to implement it: please use the siunitx package for units (ex: \SI{3.09}{\metre}) so everything is spaced perfectly and is really simple to customise things like the decimal separator ect.

**Comment by Proektsoftbg** [comment]
@scopeofaperture Good idea! Thank you for suggesting that. We will consider it when we start.

**Comment by mihdicaballero** [comment]
This would be a great improvement! LaTeX exporting could give the ability to append the calculations as annexes in a report without going to word and convert to LaTeX or append it as a PDF with other format. With LaTeX you could use the same class file with header and footer design. And also have a complete table of contents for the report.

**Comment by Proektsoftbg** [comment]
Hi!

Thanks to Adam Rawlinson, we have an easy workaround for the export from Calcpad към LaTeX ([https://www.latex-project.org](https://www.latex-project.org)).

1. You need to install Pandoc first ([https://pandoc.org](https://pandoc.org)). 
2. Then, export you results from Calcad to docx.
3. Convert the docx file to tex with the following command:
`pandoc -f docx -t latex  input.docx" -o  output.tex`

Please, note that pandoc installs in "C:\Users\Ned\AppData\Local\Pandoc" if you use the msi installer.

The results look very well. I tried the Cubic Equation Example:(https://github.com/Proektsoftbg/Calcpad/blob/main/Examples/Mathematics/Cubic%20Equation.cpd)

This is how it looks in Overleaf ([https://www.overleaf.com](https://www.overleaf.com)):

![X2F38v0V9wCPlVNh](https://user-images.githubusercontent.com/90918363/233998755-64fa27ef-48b5-4c65-bc17-fa56b4acda35.png)

Attached here are the test files, included cpd, docx, tex and pdf versions:
[Cubic Equation.zip](https://github.com/Proektsoftbg/Calcpad/files/11310527/Cubic.Equation.zip)

Pandoc is also free and open source, so we can connect it to Calcpad in future and add a button to automate the conversion. At least until we write our own LaTeX export. (It will not be a big problem, since it follows the same logic as the Html and OpenXml ones).

**Comment by xyont** [comment]
Alternatively, using [OpenOffice](https://extensions.openoffice.org/en/project/writer2latex)

**Comment by Proektsoftbg** [comment]
@xyont . Thank you for sharing that.

**Comment by xyont** [comment]
Hi, 

Previously at my old personal project of Python/C++ codes generate plain text ASCII file output only. Then i modified to be capable to output as LaTeX format.

It seems does not need any external software or library dependencies, make it less complex by direct converting Calcpad cpd files internally. Some mapping is required for Greek symbol, vector graphics or bitmap can be inserted or convert to LaTeX drawing also.

![2009-12-07_010647](https://github.com/Proektsoftbg/Calcpad/assets/72367311/fea7558d-71da-410d-bdfc-8818ac3ab10d)
